presto-accumulo/src/main/java/com/facebook/presto/accumulo/model/Row.java

```diff
@@ -0,0 +1,246 @@
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.accumulo.AccumuloErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.VALIDATION;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.TimeType.TIME;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Class to contain an entire Presto row, made up of {@link Field} objects.
+ * <p>
+ * Used by {@link AccumuloPageSink} for writing data as well as the
+ * test cases.
+ */
+public class Row
+{
+    private List<Field> fields = new ArrayList<>();
+
+    /**
+     * Creates a new instance of {@link Row}.
+     */
+    public Row()
+    {}
+
+    /**
+     * Copy constructor from one Row to another
+     *
+     * @param row Row, copied
+     */
+    public Row(Row row)
+    {
+        requireNonNull(row, "row is null");
+        for (Field f : row.fields) {
+            fields.add(new Field(f));
+        }
+    }
+
+    /**
+     * Appends the given field to the end of the row
+     *
+     * @param f Field to append
+     * @return this, for fluent programming
+     */
+    public Row addField(Field f)
+    {
+        fields.add(f);
+        return this;
+    }
+
+    /**
+     * Appends the a new {@link Field} of the given object and type to the end of the row
+     *
+     * @param v Value of the field
+     * @param t Type of the field
+     * @return this, for fluent programming
+     */
+    public Row addField(Object v, Type t)
+    {
+        fields.add(new Field(v, t));
+        return this;
+    }
+
+    /**
+     * Gets the field at the given index
+     *
+     * @param i Index in the row to retrieve
+     * @return Field
+     * @throws IndexOutOfBoundsException If the index is out of bounds
+     */
+    public Field getField(int i)
+    {
+        return fields.get(i);
+    }
+
+    /**
+     * Gets a list of all internal fields. Any changes to this list will affect this row.
+     *
+     * @return List of fields
+     */
+    public List<Field> getFields()
+    {
+        return fields;
+    }
+
+    /**
+     * Gets the length of the row, i.e. number of fields
+     *
+     * @return Length
+     */
+    public int length()
+    {
+        return fields.size();
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Arrays.hashCode(fields.toArray());
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj instanceof Row) {
+            Row r = (Row) obj;
+            int i = 0;
+            for (Field f : r.getFields()) {
```

Comment:
You can call `Objects.equals` on lists. It will call `equals` on corresponding elements.